### ISSUE TITLE
module-sales/Model/Order/Creditmemo/Total/Tax.php division by 0 on line 139 and 140

### ISSUE THREAD

**Issue by Naberdz** [magento/magento2#37686]
### Preconditions and environment

- Magento version 2.4.6
Custom shipping method with 0 costs.

### Steps to reproduce

Create order and choose delivery method with 0 costs.
Complete order and click on creditmemo button. you will get error
<img width="514" alt="Schermafbeelding 2023-06-29 om 08 25 55" src="https://github.com/magento/magento2/assets/37579149/7cec9463-1e6d-417b-b841-30d759d8698b">

### Expected result

Creditmemo should be working

### Actual result

We get error.

had to replace those lines with the following:
```
139                 $part = $creditmemo->getShippingAmount() / ($orderShippingAmount != 0 ? $orderShippingAmount :  1);
140                 $basePart = $creditmemo->getBaseShippingAmount() / ($baseOrderShippingAmount !=0 ? $baseOrderShippingAmount : 1);
```
<img width="672" alt="Schermafbeelding 2023-06-29 om 08 26 44" src="https://github.com/magento/magento2/assets/37579149/4d790a46-e690-486d-b424-f3ef8c0d64e3">

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hi @Naberdz,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.


**Steps to reproduce**

- Create order and choose delivery method with 0 costs.
- Complete order and click on creditmemo button.


We are able to create the credit memo for the order having $0 shipping method.


<img width="1425" alt="Screenshot 2023-06-30 at 12 11 25 PM" src="https://github.com/magento/magento2/assets/51680745/720bcd0a-1eb1-42c4-bd89-c41efc0acfef">



Kindly recheck the behaviour on Magento 2.4-develop instance and elaborate steps to reproduce if the issue is still reproducible.


Thanks.

**Comment by engcom-Bravo** [comment]
Hi @Naberdz,

Kindly provide latest update on this comment https://github.com/magento/magento2/issues/37686#issuecomment-1614203759 if you are able to reproduce the issue please elaborate steps to reproduce if the issue is still reproducible. 

Thanks.

**Comment by engcom-Bravo** [comment]
@dudzio12 Thanks for your contribution & collaboration over here.

Hi @Naberdz,

This issue has been fixed under this commit d32646f.Hence we are closing this issue.

Thanks.

**Comment by engcom-Bravo** [comment]
@dudzio12 Thanks for your contribution & collaboration over here.

Hi @Naberdz,

This issue has been fixed under this commit d32646f Hence we are closing this issue.

Thanks.
